### PR TITLE
TEP-0090: Support both `matrix` and `params` in a `PipelineTask`

### PR DESCRIPTION
**Pull request by jerop**
# Changes

[TEP-0090: Matrix][tep-0090] proposed executing a `PipelineTask` in parallel `TaskRuns` and `Runs` with substitutions from combinations of `Parameters` in a `Matrix`.

This change implements the fan out of `PipelineTasks` that have both `matrix` and `params` specified. The `matrix` is used to fan out the `PipelineTask` and the `params` are the same in all the `TaskRuns`.

[tep-0090]: https://github.com/tektoncd/community/blob/main/teps/0090-matrix.md

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Release notes block below has been filled in (if there are no user facing changes, use release note "NONE")

# Release Notes

```release-note
Users can specify both `matrix` and `params` fields. The `matrix` is used to fan out the `PipelineTask` and the `params` are the same in all the `TaskRuns`.
```

### CODE REVIEW

**Comment by jerop** [comment]
cc @abayer @dibyom @lbernick please take a look

File: pkg/reconciler/pipelinerun/pipelinerun_test.go [hunk 1]
```diff
@@ -7513,11 +7513,12 @@ spec:
   params:
     - name: platform
     - name: browser
+    - name: version
   steps:
     - name: echo
       image: alpine
       script: |
-        echo "$(params.platform) and $(params.browser)"
+        echo "$(params.platform) and $(params.browser)" and $(params.version)"
```
Comment (abayer) [inline pkg/reconciler/pipelinerun/pipelinerun_test.go:7521]: I don't think it matters in practice, but there is an extra double-quote here.

**Review by abayer — APPROVED** [review]

File: pkg/reconciler/pipelinerun/pipelinerun_test.go (hunk 1 above)
Comment (jerop) [reply pkg/reconciler/pipelinerun/pipelinerun_test.go]: good catch, fixed it - thanks!

**Comment by abayer** [comment]
/test pull-tekton-pipeline-integration-tests

Job hung. =)

**Comment by jerop** [comment]
thanks @abayer 😀

**Pull request merged**